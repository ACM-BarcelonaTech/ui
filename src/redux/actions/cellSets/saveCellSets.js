import getApiEndpoint from '../../../utils/apiEndpoint';
import { CELL_SETS_SAVE } from '../../actionTypes/cellSets';
import pushNotificationMessage from '../notifications/pushNotificationMessage';
import composeTree from '../../../utils/composeTree';
import { cacheFetch } from '../../../utils/cacheRequest';

const saveCellSets = (experimentId) => async (dispatch, getState) => {
  const {
    loading, hierarchy, properties, error,
  } = getState().cellSets;


  // If no loaded data exists for our cell sets and some event would
  // trigger a push, we do not want to overwrite our cell sets with
  // empty data.
  if (loading || error) {
    return null;
  }

  const treeData = composeTree(hierarchy, properties);
  try {
    const json = await cacheFetch(
      `${getApiEndpoint()}/v1/experiments/${experimentId}/cellSets`,
      {
        method: 'PUT',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify(
          treeData,
          (k, v) => ((k === 'title') ? undefined : v),
        ),
      },
    );
    dispatch({
      type: CELL_SETS_SAVE,
      payload: {
        experimentId,
        saved: json,
      },
    });
  } catch (e) {
    dispatch(pushNotificationMessage('error', 'Could not connect to the server. Check your internet connection.', 5));
  }
};

export default saveCellSets;
