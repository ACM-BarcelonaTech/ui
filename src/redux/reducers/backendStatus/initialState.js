const initialExperimentBackendStatus = {
  loading: false,
  error: false,
  status: {
  },
};

const initialState = {};

export default initialState;

export { initialExperimentBackendStatus };
