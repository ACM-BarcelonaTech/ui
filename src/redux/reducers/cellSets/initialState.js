const initialState = {
  properties: {},
  hierarchy: [],
  selected: [],
  loading: true,
  error: false,
};

export default initialState;
