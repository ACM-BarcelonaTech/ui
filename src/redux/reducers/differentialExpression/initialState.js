const initialState = {
  properties: {
    data: [],
    comparisonGroup: {},
    comparisonType: null,
    loading: false,
    error: false,
    total: 0,
  },
  comparison: {
    type: 'within',
    advancedFilters: [],
    group: {
      between: {
        cellSet: null,
        compareWith: null,
        basis: null,
      },
      within: {
        cellSet: null,
        compareWith: null,
        basis: null,
      },
    },
    ordering: {
      orderBy: 'logFC',
      orderDirection: 'DESC',
    },
  },
};

export default initialState;
