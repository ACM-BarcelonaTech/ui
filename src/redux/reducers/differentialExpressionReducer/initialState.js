const initialState = {
  properties: {
    data: {},
    cellSets: {},
    loading: false,
    error: false,
  },
};

export default initialState;
