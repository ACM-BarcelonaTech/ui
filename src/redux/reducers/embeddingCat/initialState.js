const initialState = {
  width: 700,
  height: 550,
  pointSize: 5,
  toggleInvert: '#FFFFFF',
  masterColour: '#000000',
  umap1Domain: null,
  umap2Domain: null,
  titleText: '',
  titleSize: 20,
  titleAnchor: 'start',
  axisTitlesize: 13,
  axisTicks: 13,
  transGrid: 0,
  axesOffset: 10,
  masterFont: 'sans-serif',
  xaxisText: 'UMAP 1',
  yaxisText: 'UMAP 2',
  pointStyle: 'circle',
  pointOpa: 5,
  g1Color: 'red',
  g2mColor: 'green',
  sColor: 'blue',
  legendTextColor: '#FFFFFF',
  legendEnabled: true,
  legend: null,
  geneexpLegendloc: '',
  labelSize: 28,
  labelShow: 1,
  labelFont: 2,
  labelsEnabled: true,
  selectedClusters: [],
  testVar: null,
};

export default initialState;
