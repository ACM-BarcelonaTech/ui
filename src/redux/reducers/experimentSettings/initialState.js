const metaInitialState = {
  loading: false,
  completingStepError: false,
  loadingSettingsError: false,
  selectedConfigureEmbeddingPlot: 'cellCluster',
  changedQCFilters: new Set(),
};

const initialState = {
  info: {
    experimentId: null,
    experimentName: null,
    sampleIds: [],
    paramsHash: '',
  },
  processing: {
    meta: metaInitialState,
  },
  originalProcessing: {},
};

export { metaInitialState };
export default initialState;
