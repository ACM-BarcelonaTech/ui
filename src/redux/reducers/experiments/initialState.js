const experimentTemplate = {
  projectUuid: null,
  name: null,
  description: 'Analysis description',
  id: null,
  createdDate: null,
  lastViewed: null,
  notifyByEmail: false,
  meta: { organism: null, type: '10x' },
  sampleIds: [],
};

const initialState = {
  meta: {
    loading: false,
    error: false,
    saving: false,
  },
  ids: [],
};

export default initialState;
export { experimentTemplate };
