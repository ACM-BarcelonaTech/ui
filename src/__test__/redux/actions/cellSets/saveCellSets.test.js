import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import fetchMock, { enableFetchMocks } from 'jest-fetch-mock';
import saveCellSets from '../../../../redux/actions/cellSets/saveCellSets';
import initialState from '../../../../redux/reducers/cellSets/initialState';

jest.mock('localforage');

enableFetchMocks();
const mockStore = configureStore([thunk]);

describe('saveCellSets action', () => {
  const experimentId = '1234';

  const hierarchy = [{ key: 'root', children: [{ key: 'child1' }, { key: 'child2' }] }];
  const properties = {
    root: { name: 'root node', rootNode: true },
    child1: { name: 'child 1', color: '#ffff00' },
    child2: { name: 'child 2', color: '#ff00ff' },
  };

  beforeEach(() => {
    const response = new Response(JSON.stringify({}));

    fetchMock.resetMocks();
    fetchMock.doMock();
    fetchMock.mockResolvedValueOnce(response);
  });

  it('Does not dispatch on loading state', async () => {
    const store = mockStore({ cellSets: { loading: true, error: false } });
    store.dispatch(saveCellSets(experimentId));
    expect(store.getActions().length).toEqual(0);
  });

  it('Does not dispatch on error state', async () => {
    const store = mockStore({ cellSets: { error: true, loading: false } });
    store.dispatch(saveCellSets(experimentId));
    expect(store.getActions().length).toEqual(0);
  });

  it('Dispatches a saved action when run correctly.', async () => {
    const store = mockStore({
      cellSets: {
        ...initialState, loading: false, hierarchy, properties,
      },
    });
    store.dispatch(saveCellSets(experimentId));

    const firstAction = store.getActions()[0];

    expect(firstAction).toMatchSnapshot();
  });

  it('Dispatches a notification when fetch fails.', async () => {
    const store = mockStore({
      cellSets: {
        ...initialState, loading: false, hierarchy, properties,
      },
    });

    fetchMock.resetMocks();
    fetchMock.mockReject(new Error('some weird error that happened'));
    store.dispatch(saveCellSets(experimentId));

    const firstAction = store.getActions()[0];
    expect(firstAction).toMatchSnapshot();
  });
});
