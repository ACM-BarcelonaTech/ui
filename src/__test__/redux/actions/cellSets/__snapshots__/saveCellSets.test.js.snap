// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`saveCellSets action Dispatches a notification when fetch fails. 1`] = `undefined`;

exports[`saveCellSets action Dispatches a saved action when run correctly. 1`] = `
Object {
  "payload": Object {
    "experimentId": "1234",
    "saved": Object {
      "one": "one",
    },
  },
  "type": "cellSets/save",
}
`;
